app/scripts/app-init.js

```diff
@@ -46,5 +64,14 @@ function importAllScripts() {
   );
 }
 
-// Placing script import call here ensures that scripts are inported each time service worker is activated.
-importAllScripts();
+// eslint-disable-next-line no-undef
+self.addEventListener('install', importAllScripts);
+
+/*
+ * Message event listener below loads script if they are no longer available.
+ * chrome below needs to be replaced by cross-browser object,
+ * but there is issue in importing webextension-polyfill into service worker.
+ * chrome does seems to work in at-least all chromium based browsers
+ */
+// eslint-disable-next-line no-undef
+chrome.runtime.onMessage.addListener(importAllScripts);
```

Comment:
Hey @naugtur : this message listener is added to revive service worker as it becomes inactive. As service worker becomes inactive the background files are no longer available and we need to reload them.

Reply:
If scriptsLoaded is true, it doesn't do anything. So why wait to load the scripts if they can be installed once, synchronously? What do we gain from doing it either on install or on message instead of always and right away?

Does it have anything to do with google folks suggesting we move importScripts to 'install' event?

Reply:
I was wondering the same thing as you @naugtur. It appears that when I manually terminate the service worker, `scriptsLoadedInitiated` gets reset. This implies to me that `app-init.js` is torn down and reloaded. The app continues to run throughout this process. 

However, after initially importing the scripts and then terminating the service worker, I'm running into this error (see the second error). I believe this error is related to concerns in this thread:
<img width="948" alt="Screenshot 2022-07-23 at 2 49 56 AM" src="https://user-images.githubusercontent.com/20778143/180505732-e99d8780-1d86-40ec-b5f2-725bc8883f94.png">

Reply:
Good point @naugtur , @digiwand : I have updated the branch plz check.